bin/applab-data-stats

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env ruby
+
+require 'aws-sdk'
+require_relative '../deployment'
+require 'cdo/db'
+require_relative '../shared/middleware/helpers/storage_id'
+
+db = Aws::DynamoDB::Client.new
+
+last_evaluated_key = nil
+hash_map = {}
+channel_map = {}
+rows = 0
+consumed_capacity = 0
+
+begin
+  response = db.scan(
+      table_name: CDO.dynamo_tables_table,
+      attributes_to_get: ['channel_id', 'hash'],
+      exclusive_start_key: last_evaluated_key,
+      return_consumed_capacity: 'TOTAL'
+  )
+  consumed_capacity += response.consumed_capacity.capacity_units
+  page = response.first
+
+  page[:items].each do |item|
+    channel_map[item['channel_id']] = 1
+    hash_map[item['hash']] = 1
+    rows += 1
+  end
+
+  last_evaluated_key = page[:last_evaluated_key]
+end while last_evaluated_key
+
+$stderr.puts "tables: #{hash_map.keys.length}"
+$stderr.puts "channels: #{channel_map.keys.length}"
+$stderr.puts "rows: #{rows}"
+$stderr.puts "1st pass consumed_capacity: #{consumed_capacity}"
+
+consumed_capacity = 0
+
+def get_item_size(item)
+  $stderr.puts "bogus item #{item}" unless item['table_name']
+  table_name_length = item['table_name'] ? item['table_name'].length : 0
+  item['value'].length + 2 * table_name_length + 130
+end
+
+$channel_map = {}
```

Comment:
`channel_map` just keeps track of existence of channels in the 1st pass. I initially tried to reuse it in the second pass, before ruby scoping rules made me change it to a global. I've given them distinct names now.